Upgrade to repo-config v0.3.0

llucax opened the pull request:
- Update to repo-config v0.3.0
- Fix copy&paste error in URL

llucax committed:
Update to repo-config v0.3.0

This also regenerates the cookiecutter templates, so a lot of files got
updated.

The main improvements are:

- Add documentation website generation
- Add support for merge queues
- Add cookiecutter replay file to easy future upgrades of repo-config
- Rename issue templates to follow the repo-config names
- Improve README
- Improve CONTRIBUTING
- Add some extra common configuration for linting tools

Signed-off-by: Leandro Lucarella <luca-frequenz@llucax.com>

llucax committed:
Fix copy&paste error in URL

It was pointing to the SDK instead of this repo.

Signed-off-by: Leandro Lucarella <luca-frequenz@llucax.com>

llucax commented:
Converted to draft because I'm trying out a protolint github action.

tiyash-basu-frequenz reviewed (commented):
~Just noticed that this is a draft PR. Nonetheless, submitting an early feedback.~

Looks like I did not add my comment here. Sorry, please ignore it.

tiyash-basu-frequenz commented on .github/workflows/ci.yaml:
```diff
@@ -32,19 +33,8 @@ jobs:
-
       - name: Run protolint
-        run: protolint lint proto
+        uses: plexsystems/protolint-action@v0.7.0
```
The last commit in https://github.com/plexsystems/protolint-action is more than one year ago, and it uses protolint version 0.37.0, which is really old now. I'd say that while this provides syntactic sugar, it is not worth it. I'd suggest just upgrading protolint package in the existing lines to 0.45.0.

llucax replied:
Ah, good observation. My main goal to look for preexisting actions was to provide inline comments on PRs breaking the linting, which I find super useful. But it seems like this one doesn't do that either. There are a couple of other alternatives, I'm trying one now that supports this (and is done by the same author as protolint).

llucax replied:
![image](https://github.com/frequenz-floss/frequenz-api-common/assets/1031485/eeeb50bb-4c8a-48df-bdf2-469373455aff)

llucax replied:
BTW, I'm just toying around with this, probably not the best idea to mix it with the update of repo-config, sorry about that, I should have created a separate PR for thus, was just tempted when trying to identify stuff that is missing in repo-config for the next release.

llucax replied:
I created a separate PR for this. Again, sorry about the noise! :)

This is ready for review now.

tiyash-basu-frequenz replied:
:rocket:

llucax committed:
Only publish docs (and releases) if protolint pass

We don't want to move forward with releases and publishing of docs if
the proto files are not clean.

Signed-off-by: Leandro Lucarella <luca-frequenz@llucax.com>

llucax committed:
Add a nice name to the protolint job

Signed-off-by: Leandro Lucarella <luca-frequenz@llucax.com>

llucax commented:
Enabling auto-merge

llucax commented:
I also enabled merge queues for this repo now.

llucax commented:
Ping!

llucax commented:
Thanks @Marenz, but we need someone from the @frequenz-floss/api-team to approve here.

tiyash-basu-frequenz commented on .github/RELEASE_NOTES.template.md:
```diff
@@ -1,12 +1,12 @@
-# Frequenz Common API Release Notes
+# Frequenz common API Release Notes
```
Being pedantic, but I'd suggest `Common` with `C` in upper case. It would make it more consistent with the rest of the words in the heading.
Also, I think it's better if we treat `Frequenz Common API` as a proper name, and capitalize the first letter of every word in it, in general.

llucax replied:
Agreed, but this would have to be ideally fixed in repo-config to avoid having to correct it each time we update the templates.

tiyash-basu-frequenz replied:
Makes sense. Do we have an issue for this in the repo-config repo?

llucax replied:
* https://github.com/frequenz-floss/frequenz-repo-config-python/pull/89

Will fix this PR too.

tiyash-basu-frequenz replied:
Thanks!

llucax replied:
Updated

tiyash-basu-frequenz replied:
Thanks!

tiyash-basu-frequenz commented on CONTRIBUTING.md:
```diff
@@ -1,9 +1,6 @@
-Contributing to Frequenz Common API
-===================================
+# Contributing to Frequenz common API
```
In line with my other comment: `Frequenz Common API` (upper case `C`)?

tiyash-basu-frequenz commented on README.md:
```diff
@@ -1,2 +1,14 @@
-# frequenz-api-common
-Common definitions for frequenz GRPC APIs
+# Frequenz common API
```
In line with my other comment: `Frequenz Common API` (upper case `C`)?

tiyash-basu-frequenz commented on RELEASE_NOTES.md:
```diff
@@ -1,4 +1,4 @@
-# Frequenz Common API Release Notes
+# Frequenz common API Release Notes
```
In line with my other comment: `Frequenz Common API` (upper case `C`)?

tiyash-basu-frequenz commented on .github/workflows/ci.yaml:
```diff
@@ -144,7 +144,7 @@ jobs:
   publish-docs:
     name: Publish documentation website to GitHub pages
-    needs: ["nox", "build"]
+    needs: ["nox", "build", "protolint"]
```
This check makes a lot of sense. Would it make sense to add the `protolint` job as a pre-requisite to `build` job? It looks like the build job need not run if we know the protolint check has failed.

llucax replied:
I think it is good to have the build run even if linting fails, so we can detect issues in both building and linting at the same time, if that is the case. If we add a dependency there then we'll first get a linting error, we'll fix it, push again only to find out the build fails too.
